File: split-a-string-in-balanced-strings.rb
# @param {String} s
# @return {Integer}
def balanced_string_split(s)
  p s.scan(/(RL|LR)/)
end

# balanced_string_split('RLRRLLRLRL')
# balanced_string_split('RLLLLRRRLR')
# balanced_string_split('LLLLRRRR')
balanced_string_split('RLRRRLLRLL')

File: php/split-a-string-in-balanced-strings.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function balancedStringSplit($s) {
    $balance=0;
    $count=0;
    for ($i=0; $i < strlen($s); $i++) 
    { 
        if ($s[$i]=='L') {
            $balance++;
        }
        else {
            $balance--;
        }
        if ($balance==0) {
            $count++;
        }
    }
    return $count;
    }
}
$solution = new Solution();
$solution->balancedStringSplit('RLRRRLLRLL');
